docs/G-Codes.md

```diff
@@ -754,3 +754,11 @@ is enabled (also see the
   defaults to the current time in "YYYYMMDD_HHMMSS" format. Note that
   the suggested input shaper parameters can be persisted in the config
   by issuing `SAVE_CONFIG` command.
+
+### XPT 2046 touch screen controller commands
+The following command is available when a
+[xpt_2046 config section](Config_Reference.md#xpt_2046)
+is enabled.
+- `XPT_DEBUG [ENABLE=[1|0]]`: Enables or disables debug reporting. When
+  enabled, all touch events and the associated coordinates are printed to the
+  terminal.
```

Comment:
FWIW, I would generally prefer not to add debugging commands to Klipper (they complicate the documentation and very few people will ultimately use them).  If debugging output is to be produced, best to write it to the klipper log and not the gcode stream (newer interfaces, such as mainsail/fluidd) don't really use the gcode terminal).

Reply:
``<rant>``
... which is a crying shame. I use mainsail, and the reverse console bugs me greatly, especially since I constantly need to use it to work around missing functionality (or bugs?) of mainsail. It is one of the reasons why I'm not at all sure it's a keeper. Web interfaces are supposed to make one's life easier, not to actively make things harder for everyone whose solution doesn't fit the generic mold. Fluidd, I'm thinking of you with your silly insistance of shipping only rasbian images. 
``</rant>``

That said, this is required for calibration, and should probably be reworded as such. The coordinate system of these touchpads are unrelated to the coordinate system of the underlying display, component tolerances are significant, and some non-linearity is to be expected. 

I've found that the best way to calibrate the touch screen works like this:

1. define a button on the screen
2. enable the reporting, 
3. Touch all 4 corners with a stylus. 
4. Copy the reported coordinates to the config file. 

This would probably work with the logfile as well, but finding back data is a lot harder because you don't have any frame of reference.

Reply:
Okay.  If the command is used for calibration then that's fine.